client/landing/gutenboarding/index.tsx

```diff
@@ -144,3 +150,28 @@ function waitForSelectedSite(): Promise< Site | undefined > {
 		select( SITE_STORE ).getSite( selectedSite );
 	} ).finally( unsubscribe );
 }
+/**
+ * If available-designs-config.json has been updated, replace the cached list
+ * of designs with the updated designs
+ */
+function checkIfRandomizedDesignsAreUpToDate() {
+	const designsInStore = select( ONBOARD_STORE ).getRandomizedDesigns();
+	if ( ! isUpToDate( designsInStore.featured, availableDesigns.featured ) ) {
+		dispatch( ONBOARD_STORE ).setRandomizedDesigns( {
+			...availableDesigns,
+			featured: shuffle( availableDesigns.featured ),
+		} );
+	}
+}
+
+/**
+ *
+ * Compare cached designs in the ONBOARD_STORE to the source of designs in
+ * available-designs-config.json
+ *
+ * @param stored randomizedDesigns cached in WP_ONBOARD
+ * @param available designs sourced from available-designs-config.json
+ */
+function isUpToDate( stored: Design[], available: Design[] ): boolean {
+	return isEmpty( xorWith( stored, available, isEqual ) );
```

Comment:
Thanks for reviewing @alshakero! For this function, I think we'll still need to do a deep comparison to protect the data integrity of any of the design objects. My thinking was that if we change anything in `available-designs-config.json` we'll want the cache to be invalidated so that someone with persisted state doesn't have incorrect values. E.g. if we swap template, theme, or categories, or accidentally ship a design with a typo in one of these fields, we'll want to make sure that a follow-up deploy will reach users with persisted state.

Reply:
oooh totallyl missed that! Was just thinking about adding or deleting whole designs.